tests/models/test_openai.py

```diff
@@ -640,6 +640,21 @@ async def test_image_url_input(allow_model_requests: None):
     )
 
 
+@pytest.mark.vcr()
+async def test_image_as_binary_content_tool_response(
+    allow_model_requests: None, image_content: BinaryContent, openai_api_key: str
+):
+    m = OpenAIModel('gpt-4o', provider=OpenAIProvider(api_key=openai_api_key))
+    agent = Agent(m)
+
+    @agent.tool_plain
+    async def get_image() -> BinaryContent:
+        return image_content
+
+    result = await agent.run(['What fruit is in the image you can get from the get_image tool?'])
+    assert result.output == snapshot('The fruit in the image is a kiwi.')
```

Comment:
Sorry to point out this again, but I think we are more interested in the `result.all_messages()` rather than the output.

```suggestion
    assert result.all_messages() == snapshot()
```

I want to be able to see the parts with multimodal in the history.